src/python/pants/jvm/resources.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants.core.util_rules.archive import ZipBinary
+from pants.core.util_rules.source_files import SourceFilesRequest
+from pants.core.util_rules.stripped_source_files import StrippedSourceFiles
+from pants.engine.process import BashBinary, Process, ProcessResult
+from pants.engine.rules import Get, collect_rules, rule
+from pants.engine.target import SourcesField
+from pants.engine.unions import UnionRule
+from pants.jvm.compile import (
+    ClasspathEntry,
+    ClasspathEntryRequest,
+    CompileResult,
+    FallibleClasspathEntry,
+)
+from pants.jvm.target_types import JvmResourcesFieldSet, JvmResourcesGeneratorFieldSet
+
+
+class JvmResourcesRequest(ClasspathEntryRequest):
+    field_sets = (
+        JvmResourcesFieldSet,
+        JvmResourcesGeneratorFieldSet,
+    )
+
+
+_PANTS_RESOURCES_PARTIAL_JAR_FILENAME = "pants-resources.jar"
```

Comment:
Unsure whether we need individual names for each resources JAR, or if I should come up with a name (and if so, what that name should be)

Reply:
I think that makes sense here too. Thanks!